### PR TITLE
Create Language Preference

### PR DESCRIPTION
**Pull request by shinnosuke14202**
## Related Tickets
- [#Task 78733: Content](https://edu-redmine.sun-asterisk.vn/issues/78733)
## WHAT
- Create language preference
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

**Review by nang2002hv — APPROVED** [review]

**Review by nhikiu — APPROVED** [review]

File: app/src/main/java/com/example/book/screen/home/HomeFragment.kt [hunk 1]
```diff
@@ -1,35 +1,68 @@
...
 class HomeFragment : BaseFragment<FragmentHomeBinding>(), HomeContract.View {
     private lateinit var homePresenter: HomePresenter
     private val homeBookAdapter: HomeBooksAdapter by lazy { HomeBooksAdapter() }
     private val novelsAdapter: HomeBooksAdapter by lazy { HomeBooksAdapter() }
+    private val popUpBinding by lazy {
+        PopUpLanguagesBinding.inflate(layoutInflater)
+    }
+    private lateinit var sharedPreferences: SharedPreferences
+    private lateinit var editor: SharedPreferences.Editor
+    private lateinit var dialog: Dialog
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/home/HomeFragment.kt]: Sắp xếp lại thứ tự đi em

File: app/src/main/java/com/example/book/screen/home/HomeFragment.kt [hunk 2]
```diff
@@ -1,35 +1,68 @@
...
+        sharedPreferences = requireContext().getSharedPreferences(SETTINGS, MODE_PRIVATE)
+        editor = sharedPreferences.edit()
+        currentLanguage = sharedPreferences.getString(SELECTED_LANGUAGE, ENG)
         viewBinding.apply {
             rvHomeBooks.apply {
                 adapter = homeBookAdapter
                 layoutManager = GridLayoutManager(requireContext(), 2)
             }
             rvNovels.apply {
                 adapter = novelsAdapter
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/home/HomeFragment.kt]: quá nhiều tầng code lồng vào nhau, hạn chế sử dụng .apply khi không cần thiết em ơi

File: app/src/main/java/com/example/book/screen/home/HomeFragment.kt [hunk 3]
```diff
@@ -40,6 +73,64 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>(), HomeContract.View {
...
+                editor.apply()
+                refreshApp()
+            }
+            clEnglish.setOnClickListener {
+                editor.putString(SELECTED_LANGUAGE, ENG)
+                editor.apply()
+                refreshApp()
+            }
+        }
+    }
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/home/HomeFragment.kt]: Đây cũng vậy, quá nhiều tầng code, refactor đi em

File: app/src/main/java/com/example/book/screen/home/HomeFragment.kt [hunk 4]
```diff
@@ -40,6 +73,64 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>(), HomeContract.View {
...
+
+    private fun refreshApp() {
+        val refresh = Intent(activity, MainActivity::class.java)
+        startActivity(refresh)
+        activity?.finish()
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/home/HomeFragment.kt:130]: Clear backstack nữa em nha

**Review by ghost — CHANGES_REQUESTED** [review]
check my cmts

**Review by ghost — APPROVED** [review]
Oke

**Pull request merged**